### PR TITLE
PP-794: PBS does not update job comment while resuming a suspended job

### PR DESCRIPTION
**Pull request by nithinj**
#### Issue-ID

* **[PP-794](https://pbspro.atlassian.net/browse/PP-794)**

#### Problem description
* PBS does not update job comment while resuming a suspended job

#### Cause / Analysis
* Server was not performing job comment update while resuming the job from suspended state.

#### Solution description
* Server will update job comment while resuming from suspended state.
* PTL to cover this scenario.

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

CC: @arungrover

### CODE REVIEW

File: test/tests/functional/pbs_job_comment_on_resume.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    Testing job comment is accurate
+    """
+
+    def test_job_comment_on_resume(self):
+        """
+        Testing whether job comment is accurate
+        after resuming from suspended state.
+        """
+        attr = {'resources_available.ncpus': 1}
+        self.server.create_vnodes(
```
Comment (arungrover) [inline test/tests/functional/pbs_job_comment_on_resume.py]: This step isn't really needed. you can change resources of exiting node and just use that to run a job.

Comment (bhroam) [reply test/tests/functional/pbs_job_comment_on_resume.py]: Or just have jid1 request ncpus=8 (the default node has 8cpus).

File: src/server/req_signal.c [hunk 2]
```diff
@@ -497,6 +497,14 @@ post_signal_req(struct work_task *pwt)
 			pjob->ji_wattr[(int) JOB_ATR_resc_released_list].at_flags &= ~ATR_VFLAG_SET;
 
 			svr_setjobstate(pjob, JOB_STATE_RUNNING, JOB_SUBSTATE_RUNNING);
+
+			if (!(pjob->ji_qs.ji_svrflags & JOB_SVFLG_SubJob)) {
+				job_attr_def[(int) JOB_ATR_Comment].at_decode(
+						&pjob->ji_wattr[(int) JOB_ATR_Comment], (char *) 0,
+						(char *) 0,
+						form_attr_comment("Job run at %s",
```
Comment (bhroam) [inline src/server/req_signal.c]: You can combine this line with the line above and it'll be about the same length as the line above that.  The more lines you split over makes it harder to read.

File: test/tests/functional/pbs_job_comment_on_resume.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            usenatvnode=True)
+        attr = {'queue_type': 'execution', 'started': 't', 'enabled': 't',
+                'priority': 200}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, attr, id='expressq')
+        J = Job(TEST_USER)
+        jid1 = self.server.submit(J)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        attr = {'queue': 'expressq'}
+        J = Job(TEST_USER, attr)
+        J.set_sleep_time(2)
```
Comment (bhroam) [inline test/tests/functional/pbs_job_comment_on_resume.py]: No big deal, but you could set the sleep time to 1 for a slightly faster test.

File: test/tests/functional/pbs_job_comment_on_resume.py (hunk 1 above)
Comment (nithinj) [reply test/tests/functional/pbs_job_comment_on_resume.py]: I've seen other test cases which are requesting 8 ncpus. Though never guessed that is the default value. Thanks and I'll use the same.

Comment (nithinj) [reply test/tests/functional/pbs_job_comment_on_resume.py]: I've tried with ncpus as 8. But it's failing in my VM as the ncpus by default in it is 4 and PTL doesn't seem to set nccpus as 8 during the set-up phase. I'll modify the existing nodes ncpus to one, instead.

**Comment by nithinj** [comment]
Hi @arungrover and @bhroam ,
Thanks for reviewing it :)
I've made the suggested changes. Please check.

File: test/tests/functional/pbs_job_comment_on_resume.py (hunk 1 above)
Comment (bhroam) [reply test/tests/functional/pbs_job_comment_on_resume.py]: Interesting.  I thought PTL reverted back to 1 vnode with 8cpus and 8gb of mem.  I just looked through the revert_to_defaults() code and only found that it deletes the vnodes.  I guess it'll revert back to the real resources of the node you are running on.

**Comment by nithinj** [comment]
Thanks @bhroam for your sign-off.

Hi @arungrover ,
I've made the suggested changes in the PTL. Please review.

**Comment by arungrover** [comment]
@nithinj  change looks good. Please squash and rebase

**Comment by nithinj** [comment]
Thanks @arungrover for the sign-off. 
I've squashed and re-based my branch. Please take this further.

**Comment by arungrover** [comment]
merged

**Pull request merged**